routes/common/load_balancer_helper.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+class Routes::Common::LoadBalancerHelper < Routes::Common::Base
+  def list
+    lbs = Serializers::LoadBalancer.serialize(project.load_balancers_dataset.authorized(@user.id, "LoadBalancer:view").all, {include_path: true})
+    @app.instance_variable_set(:@lbs, lbs)
+    @app.view "networking/load_balancer/index"
+  end
+
+  def post(name: nil)
+    Authorization.authorize(@user.id, "LoadBalancer:create", project.id)
+
+    required_parameters = %w[name private_subnet_id algorithm src_port dst_port health_check_endpoint]
+    allowed_optional_parameters = %w[health_check_interval health_check_timeout health_check_up_threshold health_check_down_threshold]
+    request_body_params = Validation.validate_request_body(params, required_parameters, allowed_optional_parameters)
+
+    ps = PrivateSubnet.from_ubid(request_body_params["private_subnet_id"])
+    Authorization.authorize(@user.id, "PrivateSubnet:view", ps.id)
+
+    lb = Prog::Vnet::LoadBalancerNexus.assemble(
+      ps.id,
+      name: name,
+      algorithm: request_body_params["algorithm"],
+      src_port: request_body_params["src_port"],
+      dst_port: request_body_params["dst_port"],
+      health_check_endpoint: request_body_params["health_check_endpoint"],
+      health_check_interval: request_body_params["health_check_interval"],
+      health_check_timeout: request_body_params["health_check_timeout"],
+      health_check_up_threshold: request_body_params["health_check_up_threshold"],
+      health_check_down_threshold: request_body_params["health_check_down_threshold"]
+    ).subject
+
+    flash["notice"] = "'#{name}' is created"
+
+    @request.redirect "#{project.path}#{lb.path}"
+  end
+
+  def get
+    Authorization.authorize(@user.id, "LoadBalancer:view", @resource.id)
+    vms = @resource.private_subnet.vms
+    attached_vms = @resource.vms
+    @app.instance_variable_set(:@attachable_vms, Serializers::Vm.serialize(vms.reject { |vm| attached_vms.map(&:id).include?(vm.id) }))
```

Comment:
no need to call `.map` on `attached_vms` in all iterations
```suggestion
    attached_vm_ids = @resource.vms.map(&:id)
    @app.instance_variable_set(:@attachable_vms, Serializers::Vm.serialize(vms.reject { attached_vm_ids.include?(_1.id) }))
```